pkg/odo/util/validation/validation.go

```diff
@@ -18,5 +18,14 @@ func ValidateName(name string) error {
 	}
 
 	return nil
+}
 
+// ValidateHost validates that the provided host is a valid subdomain according to DNS (RFC 1123) rules
+func ValidateHost(host string) error {
```

Comment:
Please add a unit test

Reply:
I could but there's not much interest in doing so since it's calling a presumably well tested function in k8s apimachinery.

Reply:
OK let's skip it